MRFIXIT: ALLOWS FOR THE OVERRIDE OF THE GENERAL VIDEO RESOLUTION SETTINGS INSIDE THE RICE SETTINGS IN MUPEN64PLUS.CFG
--- a/src/Config.cpp	2018-11-09 16:39:42.608159241 +0000
+++ b/src/Config.cpp	2018-11-09 16:52:27.401663331 +0000
@@ -306,8 +306,10 @@
     /* Set default config parameters */
     ConfigSetDefaultBool(l_ConfigVideoGeneral, "Fullscreen", 0, "Use fullscreen mode if True, or windowed mode if False ");
     ConfigSetDefaultInt(l_ConfigVideoGeneral, "ScreenWidth", 640, "Width of output window or fullscreen width");
     ConfigSetDefaultInt(l_ConfigVideoGeneral, "ScreenHeight", 480, "Height of output window or fullscreen height");
+    ConfigSetDefaultInt(l_ConfigVideoRice, "ScreenWidth", 640, "Width of output window or fullscreen width");
+    ConfigSetDefaultInt(l_ConfigVideoRice, "ScreenHeight", 480, "Height of output window or fullscreen height");
     ConfigSetDefaultInt(l_ConfigVideoRice, "ScreenLeftOffset", 0, "Position of the left of the screen");
     ConfigSetDefaultBool(l_ConfigVideoGeneral, "VerticalSync", 0, "If true, activate the SDL_GL_SWAP_CONTROL attribute");
 
     ConfigSetDefaultInt(l_ConfigVideoRice, "Version", CONFIG_PARAM_VERSION, "Mupen64Plus Rice Video Plugin config parameter version number");
@@ -426,9 +426,14 @@
 static void ReadConfiguration(void)
 {
     windowSetting.bDisplayFullscreen = ConfigGetParamBool(l_ConfigVideoGeneral, "Fullscreen");
-    windowSetting.uDisplayWidth = ConfigGetParamInt(l_ConfigVideoGeneral, "ScreenWidth");
-    windowSetting.uDisplayHeight = ConfigGetParamInt(l_ConfigVideoGeneral, "ScreenHeight");
+    if (ConfigGetParamInt(l_ConfigVideoRice, "ScreenWidth") == 0 || ConfigGetParamInt(l_ConfigVideoRice, "ScreenHeight") == 0) {
+        windowSetting.uDisplayWidth = ConfigGetParamInt(l_ConfigVideoGeneral, "ScreenWidth");
+        windowSetting.uDisplayHeight = ConfigGetParamInt(l_ConfigVideoGeneral, "ScreenHeight");
+    } else {
+        windowSetting.uDisplayWidth = ConfigGetParamInt(l_ConfigVideoRice, "ScreenWidth");
+        windowSetting.uDisplayHeight = ConfigGetParamInt(l_ConfigVideoRice, "ScreenHeight");
+    }
     windowSetting.leftOffset = ConfigGetParamInt(l_ConfigVideoRice, "ScreenLeftOffset");
     windowSetting.bVerticalSync = ConfigGetParamBool(l_ConfigVideoGeneral, "VerticalSync");
 
     defaultRomOptions.N64FrameBufferEmuType = ConfigGetParamInt(l_ConfigVideoRice, "FrameBufferSetting");

